File: sdks/csharp/src/NovaBilling/Invoices/Requests/CreateCheckoutInvoicesRequest.cs
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record CreateCheckoutInvoicesRequest
{
    /// <summary>
    /// Invoice ID
    /// </summary>
    [JsonIgnore]
    public required string Id { get; set; }

    /// <summary>
    /// URL to redirect customer after payment
    /// </summary>
    [JsonPropertyName("callbackUrl")]
    public string? CallbackUrl { get; set; }

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: sdks/php/src/Invoices/Requests/CreateCheckoutInvoicesRequest.php
<?php

namespace NovaBilling\Invoices\Requests;

use NovaBilling\Core\Json\JsonSerializableType;
use NovaBilling\Core\Json\JsonProperty;

class CreateCheckoutInvoicesRequest extends JsonSerializableType
{
    /**
     * @var ?string $callbackUrl URL to redirect customer after payment
     */
    #[JsonProperty('callbackUrl')]
    public ?string $callbackUrl;

    /**
     * @param array{
     *   callbackUrl?: ?string,
     * } $values
     */
    public function __construct(
        array $values = [],
    ) {
        $this->callbackUrl = $values['callbackUrl'] ?? null;
    }
}
